examples/step-79/doc/intro.dox

```diff
@@ -0,0 +1,597 @@
+  \varepsilon(\textbf{d}_\textbf{u})\varepsilon(\textbf{c}_{\textbf{y}_1})]
+\\
+=-\int_\Gamma \textbf{d}_\textbf{u} \cdot \textbf{t} -\int_\Omega \rho^{p}
+[\lambda \nabla \cdot \textbf{y} \nabla \cdot \textbf{d}_\textbf{u} + \mu
+\varepsilon(d_\textbf{u})\varepsilon(\textbf{y}_1)]
+@f}
+Equation 2
+@f[
+  \int_\Omega  - d_\sigma c_{z_1} +d_\sigma c_{z_2}  + H(d_\sigma)c_{y_2} =
+  -\int_\Omega -d_\sigma z_1 + d_\sigma z_2 + H(d_\sigma)y_2
+@f]
+</li>
+<li> Primal Feasibility - these equations ensure the equality constraints
+are met.
+
+Equation 3
+@f{multiline}{\int_\Omega p \rho^{p-1} c_p[\lambda \nabla \cdot
+\textbf{d}_{\textbf{y}_1} \nabla \cdot \textbf{u} +  \mu
+\varepsilon(\textbf{u}) \varepsilon(\textbf{d}_{\textbf{y}_1})] +
+\rho^{p}[\lambda \nabla \cdot \textbf{d}_{\textbf{y}_1} \nabla \cdot
+\textbf{c}_\textbf{u} +  \mu  \varepsilon(\textbf{c}_\textbf{u})
+\varepsilon(\textbf{d}_{\textbf{y}_1})]
+\\
+ =-\int_\Omega \rho^{p}[\lambda \nabla \cdot \textbf{d}_{\textbf{y}_1} \nabla
+ \cdot \textbf{u} + \mu  \varepsilon(\textbf{u}) \varepsilon
+ (\textbf{d}_{\textbf{y}_1})]  + \int_\Gamma  \textbf{d}_{\textbf{y}_1}
+ \cdot \textbf{t}
+@f}
+
+    Equation 4
+@f[
+  -\int_\Omega d_{z_1}(c_\sigma - c_{s_1})=\int_\Omega d_{z_1} (\sigma - s_1)
+@f]
+
+    Equation 5
+@f[
+  -\int_\Omega d_{z_2}(-c_\sigma-c_{s_2}) = \int_\Omega d_{z_2} (1-\sigma-s_2)
+@f]
+
+    Equation 6
+@f[
+  -\int_\Omega   d_{y_2}(c_\rho - H(c_\sigma))=\int_\Omega d_{y_2}
+  (\rho - H(\sigma))
+@f]
+  <\li>
+  <li>Complementary Slackness - these equations essentially ensure the barrier
+  is met - in the final solution, we need $s^T z = 0$
+    Equation 7
+    @f[
+      \int_\Omega d_{s_1}(c_{s_1}z_1/s_1 +  c_{z_1} )=-\int_\Omega d_{s_1}
+      (z_1 - \alpha/s_1) ,\;\;\; \alpha \to 0
+    @f]
+
+    Equation 8
+    @f[
+      \int_\Omega d_{s_2} (c_{s_2}z_2/s_2 + c_{z_2} )=-\int_\Omega d_{s_2}
+      (z_2 - \alpha/s_2) ,\;\;\; \alpha \to 0
+    @f]
+
+    Dual Feasibility - Multiplier on slacks and slack variables must be kept
+    greater than 0. (This is the only part not implemented in the
+    "assemble\_block\_system()" function)
+    @f[
+      s,z \geq 0
+    @f]
+    <\li>
+</ol>
+
+
+
+<h3>Discretization</h3>
+We use a quadrilateral mesh with $Q_1$ element to discretize the displacement and
+ displacement Lagrange multiplier. Piecewise constant $DGQ_0$ elements are used
+ to discretize the density, unfiltered density, density slack variables, and
+ multipliers for the slack variables and filter constraint.
+
+<h3>Nonlinear Algorithm</h3>
+
+While most of the discussion above follows traditional and well-known approaches
+ to solving nonlinear optimization problems, it turns out that the problem is
+ actually quite difficult to solve in practice. In particular, it is quite
+ nonlinear and an important question is not just to find search directions
+ $c_\bullet$ as discussed above based on a Newton method, but that one needs to
```

Comment:
Earlier you used the notation `$c_{\{\cdot\}}$` for the search direction. Can you please make this uniform?